marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/entities/entity-search-lib.sjs

```diff
@@ -98,13 +104,19 @@ function buildPropertyMetadata(parentPropertyName, entityModel, entityName) {
       propertyMetadata["properties"] = buildPropertyMetadata(propertyMetadata["propertyPath"], entityModel, entityName);
     }
     allPropertyMetadata.push(propertyMetadata);
+
+    if(selectedProperties && selectedProperties.includes(propertyMetadata.propertyPath)) {
+      selectedPropertyMetadata.push(propertyMetadata);
+    }
   }
   return allPropertyMetadata;
 }
 
-function getPropertyValues(currentProperty, entityInstance) {
+// Helper function used by getPropertyValuesFromInstance to fetch property values from a propertyPath
+function getPropertyPathValues(currentProperty, entityInstance) {
```

Comment:
It'll help a lot to define method comments here that define currentProperty as an object along with the expected keys in that object, which I believe are propertyPath and datatype. Otherwise, I think a reader would not know if "currentProperty" is simply a property name/path or a complex object.